emacs-everywhere.el

```diff
@@ -206,28 +210,43 @@ Make sure that it will be matched by `emacs-everywhere-file-patterns'."
 ;;;###autoload
 (defun emacs-everywhere (&optional file line column)
   "Launch the emacs-everywhere frame from emacsclient."
-  (let ((app-info (emacs-everywhere-app-info)))
-    (apply #'call-process "emacsclient" nil 0 nil
-           (delq
-            nil (list
-                 (when (server-running-p)
-                   (if server-use-tcp
-                       (concat "--server-file="
-                               (shell-quote-argument
-                                (expand-file-name server-name server-auth-dir)))
-                     (concat "--socket-name="
-                             (shell-quote-argument
-                              (expand-file-name server-name server-socket-dir)))))
-                 "-c" "-F"
-                 (prin1-to-string
-                  (cons (cons 'emacs-everywhere-app app-info)
-                        emacs-everywhere-frame-parameters))
-                 (cond ((and line column) (format "+%d:%d" line column))
-                       (line              (format "+%d" line)))
-                 (or file
-                     (expand-file-name
-                      (funcall emacs-everywhere-filename-function app-info)
-                      emacs-everywhere-file-dir)))))))
+  (let* ((app-info (emacs-everywhere-app-info))
+	 (param (delq
+		 nil (list
+                      (when (server-running-p)
+			(if server-use-tcp
+			    (concat "--server-file="
+				    (if (memq system-type '(ms-dos windows-nt cygwin))
+					(expand-file-name server-name server-auth-dir)
+				      (shell-quote-argument
+				       (expand-file-name server-name server-auth-dir))))
+			  (concat "--socket-name="
+				  (if (memq system-type '(ms-dos windows-nt cygwin))
+				      (expand-file-name server-name server-auth-dir)
+				    (shell-quote-argument
+				     (expand-file-name server-name server-socket-dir)))
+				  )))
+                      "-c" "-F"
+		      ((lambda (condition arg) (if condition (concat "\""
+								     (replace-regexp-in-string "\"" "\\\\\"" arg) "\"") arg))
+		       (memq system-type '(ms-dos windows-nt cygwin))
+		       (prin1-to-string
+			(cons (cons 'emacs-everywhere-app app-info)
+                              emacs-everywhere-frame-parameters)))
+		 
+		      (cond ((and line column) (format "+%d:%d" line column))
+			    (line              (format "+%d" line)))
+
+		      (concat "\"" (or file
+				       (expand-file-name
+					(funcall emacs-everywhere-filename-function app-info)
+					emacs-everywhere-file-dir)) "\"")
+                      )))
+	 (param-string (mapconcat 'identity param " ")))
+    (pcase system-type
+      ((or 'ms-dos 'windows-nt 'cygwin) (w32-shell-execute "open" "emacsclientw" param-string 1))
+      (_ (apply #'call-process "emacsclient" nil 0 nil param))
+      )))
```

Comment:
This looks a bit messy. The parenthesis in the `let*` binding are not lined up, and the binding for `param` is *massive*. Perhaps it would be cleaner if `param` was put into its own function at this point. Also, I don't see why you're going to the effort of `(param-string (mapconcat 'identity param " "))` just for the windows case. Lastly, please don't put `)))` on its own line.

Reply:
You're right, I found that `combine-and-quote-string` would be enough. Thanks!